test/libethereum/blockchain.cpp

```diff
@@ -124,632 +85,425 @@ void doBlockchainTests(json_spirit::mValue& _v, bool _fillin)
+void overwriteUncleHeaderForTest(mObject& uncleHeaderObj, TestBlock& uncle, std::vector<TestBlock> const& uncles, std::vector<TestBlock> const& importedBlocks)
 {
-	RLPStream rlpStream;
-	rlpStream.appendList(_tObj.count("hash") > 0 ? (_tObj.size() - 1) : _tObj.size());
-
-	if (_tObj.count("parentHash"))
-		rlpStream << importByteArray(_tObj["parentHash"].get_str());
-
-	if (_tObj.count("uncleHash"))
-		rlpStream << importByteArray(_tObj["uncleHash"].get_str());
-
-	if (_tObj.count("coinbase"))
-		rlpStream << importByteArray(_tObj["coinbase"].get_str());
-
-	if (_stateRoot)
-		rlpStream << _stateRoot;
-	else if (_tObj.count("stateRoot"))
-		rlpStream << importByteArray(_tObj["stateRoot"].get_str());
+	//uncleHeaderObj - json Uncle header with additional option fields
+	//uncle			 - uncle Block to overwrite
+	//uncles		 - previously imported uncles
+	//importedBlocks - blocks already included in BlockChain
 
-	if (_tObj.count("transactionsTrie"))
-		rlpStream << importByteArray(_tObj["transactionsTrie"].get_str());
-
-	if (_tObj.count("receiptTrie"))
-		rlpStream << importByteArray(_tObj["receiptTrie"].get_str());
-
-	if (_tObj.count("bloom"))
-		rlpStream << importByteArray(_tObj["bloom"].get_str());
-
-	if (_tObj.count("difficulty"))
-		rlpStream << bigint(_tObj["difficulty"].get_str());
-
-	if (_tObj.count("number"))
-		rlpStream << bigint(_tObj["number"].get_str());
-
-	if (_tObj.count("gasLimit"))
-		rlpStream << bigint(_tObj["gasLimit"].get_str());
-
-	if (_tObj.count("gasUsed"))
-		rlpStream << bigint(_tObj["gasUsed"].get_str());
-
-	if (_tObj.count("timestamp"))
-		rlpStream << bigint(_tObj["timestamp"].get_str());
-
-	if (_tObj.count("extraData"))
-		rlpStream << fromHex(_tObj["extraData"].get_str());
-
-	if (_tObj.count("mixHash"))
-		rlpStream << importByteArray(_tObj["mixHash"].get_str());
+	if (uncleHeaderObj.count("sameAsPreviousSibling"))
+	{
+		uncleHeaderObj.erase("sameAsPreviousSibling");
+		if (uncles.size())
+			uncle = uncles.at(uncles.size() - 1);
+		else
+			cerr << "Could not create uncle sameAsPreviousSibling: there are no siblings!";
+		return;
+	}
 
-	if (_tObj.count("nonce"))
-		rlpStream << importByteArray(_tObj["nonce"].get_str());
+	if (uncleHeaderObj.count("sameAsBlock"))
+	{
+		size_t number = (size_t)toInt(uncleHeaderObj.at("sameAsBlock"));
+		uncleHeaderObj.erase("sameAsBlock");
 
-	return rlpStream.out();
-}
+		if (number < importedBlocks.size())
+			uncle = importedBlocks.at(number);
+		else
+			cerr << "Could not create uncle sameAsBlock: there are no block with number " << number;
+		return;
+	}
 
-void overwriteBlockHeader(BlockHeader& _header, mObject& _blObj, BlockHeader const& _parent)
-{
-	auto ho = _blObj["blockHeader"].get_obj();
-	if (ho.size() != 14)
+	if (uncleHeaderObj.count("sameAsPreviousBlockUncle"))
 	{
-		BlockHeader tmp = constructHeader(
-			ho.count("parentHash") ? h256(ho["parentHash"].get_str()) : _header.parentHash(),
```

Comment:
old fillers are working but I did some changes to them to ensure that they working correctly,